### ISSUE TITLE
Patch to bind user home directories

### ISSUE THREAD

**Issue by lepasserby** [netblue30/firejail#459]
Hi! 
Perhaps a silly question, but it was mentioned that --x11 is not available for root.

Would it work for cases where firejail is called like

_sudo firejail --x11 --user=non-root-user --net=br1 program-name_ 
?

Reason for sudoing: need to bind a "host" directory into the sandbox at startup (so the jailed program's output is available to certain processes at host), thus plan to have whitelist directory directive in the profile and it requires root.

program that needs to be jailed can run as non-root (hence --user=non-root-user)

**Comment by netblue30** [comment]
The order Firejail applies the commands will prevent you from doing it. This is something that will work:

Start with a naked root sandbox where you apply the bind:
`````
$ sudo firejail --noprofile --bind=t1,t2
`````
Switch to your user:
`````
# sudo netblue
`````
Start the real x11 sandbox using --force:
`````
$ firejail --force --x11 --net=eth0 xterm
`````
Or you can do all in one line:
`````
$ sudo firejail --noprofile --bind=t1,t2 sudo -u netblue firejail --force --net=eth0 --x11 xterm
`````

**Comment by lepasserby** [comment]
I assume if I want to apply secomp filters or something like that, I'd best do that in the "last" firejail (the --force one) ?

Should first firejail come with --noprofile, or could it have one?

If "naked" firejail is started with a specific profile, would the second (--force) one honor the profile (esp whitelist directive) ?

**Comment by netblue30** [comment]
Yes, seccomp is applied to the second sandbox. If you put it in the first sandbox, the second one will not start.

And yes, you can use also use a profile for the first sandbox, but don't put in anything that might prevent the second sandbox from starting, such as seccomp, caps etc. Whitelisting should work in the first profile.

**Comment by lepasserby** [comment]
Okay, thanks!

Last (maybe) question (tangentially related, but might deserve own issue)

Let's say I want to run program in a "hollow" non-persistent homedir (as achieved by --private) but want to make one specific directory in the new homedir available to host and/or persistent.

What profile and command line directives and in what order would achieve that ?

**Comment by netblue30** [comment]
You would use --whitelist. Example:
`````
$ firejail --whitelist=~/somedir
`````
will create a "hollow" /home/user directory with /home/user/somedir imported form the filesystem and it will be persistent.

**Comment by djfd** [comment]
just a couple of cents,

there is a patch allowing to mount a few home subdirectories in bind mode, thus their content is fully synchronized between a host and sandbox.

I do not know how secure it is, is it possible to escape somehow using such kind of binding or not...
Experts advises and thoughts related to this approach are very welcome!

**Comment by netblue30** [comment]
Thanks, I'll look into it.

**Comment by chiraag-nataraj** [comment]
Isn't `--force` deprecated now? So how would one do this with the newer `firejail` versions?

**Comment by tonsimple** [comment]
@netblue30 @chiraag-nataraj 

 I too have an interest in launching firejailed apps from a limited user, but in a firejail environment  that was initially set up using "root-stuff"

"--force" still works for me (version 0.9.44.8 Debian 8) but if it is "on its way out" that may cause issues for me later on so I would like to know what's best practice is for the kind of thing I'm doing

P.S.:
Also, I would like to humbly request the return of --user command line switch. I realize there are probably practical reasons for its removal, but it was so comfy

**Comment by Vincent43** [comment]
It was removed due to security concerns so I doubt it will be ever restored.

**Comment by rusty-snake** [comment]
Something like `--bind-home=s,t` would allow a lot of workarounds, I'm thinking  #3139, …
